internal/provider/resource_consul_cluster_test.go

```diff
@@ -4,29 +4,32 @@ import (
 	"context"
 	"fmt"
 	"testing"
+	"time"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
 )
 
-var consulCluster = `
-resource "hcp_consul_cluster" "test" {
-	cluster_id         = "test-consul-cluster"
+var consulClusterUniqueResourceName = fmt.Sprintf("test-%s", time.Now().Format("200601021504"))
+
+var consulCluster = fmt.Sprintf(`
+resource "hcp_consul_cluster" "%[1]s" {
+	cluster_id         = "test-consul-cluster-original"
 	hvn_id             = hcp_hvn.test.hvn_id
 	tier               = "development"
 	min_consul_version = data.hcp_consul_versions.test.recommended
 }
-`
+`, consulClusterUniqueResourceName)
 
-var updatedConsulCluster = `
-resource "hcp_consul_cluster" "test" {
-	cluster_id = "test-consul-cluster"
+var updatedConsulCluster = fmt.Sprintf(`
+resource "hcp_consul_cluster" "%[1]s" {
+	cluster_id = "test-consul-cluster-updated"
```

Comment:
We want the cluster ID to be unique. The resource names should stay the same `test`. The issue is not with Terraform having resource name collisions, it's with the Consul clusters having cluster ID collisions.
```suggestion
	cluster_id = "%[1]s"
```